docs/docs/guides/tutorials.md

```diff
@@ -0,0 +1,215 @@
+
+## Use the CLI to Extract Snippets
+
+After you have annotated your tutorial code, use the Bluehawk CLI to extract
+code snippets. The process is the same as when 
+[extracting code examples](code-snippets#use-the-cli-to-extract-snippets), 
+because you're doing the same thing - you just pass an additional `--state`
+flag to indicate which state you want in the code example.
+
+As with extracting code examples, you use the `bluehawk snip` command,
+and you can 
+[extract code from a single file](code-snippets#extract-code-from-a-single-file) 
+or [from a directory](code-snippets#extract-code-in-a-directory). But you
+must run the command more than once, and each time you pass it the `state`
+identifier whose code example you want to generate. 
+
+For example, for our annotated code above, we might use this command to
+generate code for the `start` state:
+
+```shell
+bluehawk snip -o source/tutorial/generated/code/start/ tutorial/rn/providers/TasksProvider.js --state=start
+```
+
+And then use this command to generate code for the `final` state:
+
+```shell
+bluehawk snip -o source/tutorial/generated/code/final/ tutorial/rn/providers/TasksProvider.js --state=final
+```
+
+The input file is the same in both cases. The output directory is different.
+Every output file follows the same 
+[naming convention](code-snippets#output-file-names), and there is 
+nothing in the file name to indicate the state that was used to generate
+that file. In the example above, the generated file name would be 
+`TasksProvider.codeblock.clean-up.js` in both cases.
+
+## Include the Snippets In Your Documentation
+
+After annotating your tutorial code and extracting code examples with the 
+Bluehawk CLI, you've got code files in output directories.
+
+Now it's time to include those code files in your documentation.
+How you do that depends on your documentation tooling. For info
+about how the Realm docs team includes code examples in our documentation, see:
+[Include Code Snippets In Your documentation](code-snippets#include-the-snippets-in-your-documentation).
+
+### Automatically Update Tutorial Code
+
+For our tutorial code, we do something extra. Our tutorial applications each
+live in clonable GitHub artifact repositories so developers can clone and 
+run our tutorials. We have a 
+[GitHub Workflow](https://github.com/mongodb/docs-realm/blob/master/.github/workflows/push-to-artifact-repos.yml)
+that manages updating these tutorial repositories; we don't push to these
+repositories directly.
+
+Instead, when we push to the `tutorial` directory in our documentation 
+repository, this GitHub workflow does a few things:
+
+- It runs `bluehawk check` to [verify there are no Bluehawk errors](/reference/cli#check) 
+in our files.
+- It uses a [GitHub Action](https://github.com/mongodb/docs-realm/tree/master/.github/actions/push-to-artifact-repo) 
+to copy, commit, and push transformed code to artifact repos.
```

Comment:
It actually also uses a Bluehawk plugin for the git stuff: https://github.com/cbush/bluehawk-plugin-git